flaskr.py

```diff
@@ -94,52 +102,34 @@ def logout():
+        'select sort_order from entries where id = (?)',
+        (int(entry_id),)
+    ).fetchone()[0]
     if fnmatch.fnmatch(direction, 'up'):
-        if data[0][1] == int(entry_id):
-            flash('The entry is already on the top')
-            return redirect(url_for('show_entries'))
-        else:
-            for i, row in enumerate(data):
-                if row[1] == int(entry_id):
-                        current_row = data[i]
-                        previous_row = data[i - 1]
-                        db.execute('''UPDATE entries
-                                      SET id = ?
-                                      WHERE id = ?''',
-                                   (99999999, current_row[1]))
-                        db.execute('''UPDATE entries
-                                      SET id = ?
-                                      WHERE id = ?''',
-                                   (current_row[1], previous_row[1]))
-                        db.execute('''UPDATE entries
-                                      SET id = ?
-                                      WHERE id = ?''',
-                                   (previous_row[1], 99999999))
+        another_entry_id_so = db.execute(
+            'select id, min(sort_order) from entries where sort_order > (?)',
+            (moving_entry_so,)
+        ).fetchone()
     elif fnmatch.fnmatch(direction, 'down'):
-        if data[-1][1] == int(entry_id):
-            flash('The entry is the last one, can\'t move down')
-            return redirect(url_for('show_entries'))
-        else:
-            for i, row in enumerate(data):
-                if row[1] == int(entry_id):
-                    current_row = data[i]
-                    previous_row = data[i + 1]
-                    db.execute('''UPDATE entries
-                                  SET id = ?
-                                  WHERE id = ?''',
-                               (99999999, current_row[1]))
-                    db.execute('''UPDATE entries
-                                  SET id = ?
-                                  WHERE id = ?''',
-                               (current_row[1], previous_row[1]))
-                    db.execute('''UPDATE entries
-                                  SET id = ?
-                                  WHERE id = ?''',
-                               (previous_row[1], 99999999))
-    db.commit()
+        another_entry_id_so = db.execute(
+            'select id, max(sort_order) from entries where sort_order < (?)',
+            (moving_entry_so,)
+        ).fetchone()
+    try:
+        temp_id = another_entry_id_so[0]
+        temp_so = another_entry_id_so[1]
+        db.execute(
+            'UPDATE entries set sort_order = ? where id = ?',
+            (moving_entry_so, temp_id)
+        )
+        db.commit()
+        db.execute(
+            'UPDATE entries set sort_order = ? where id = ?',
+            (temp_so, entry_id)
+        )
+        db.commit()
```

Comment:
Added handling through python errors handling, please let me know if it's incorrect somehow